firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -875,14 +875,7 @@ extension BrowserViewController: WKNavigationDelegate {
                     CFNetworkErrors.cfurlErrorNotConnectedToInternet.rawValue
                 )
 
-                if isNativeErrorPageEnabled {
-                    let action = NativeErrorPageAction(networkError: error,
-                                                       windowUUID: windowUUID,
-                                                       actionType: NativeErrorPageActionType.receivedError
-                    )
-                    store.dispatch(action)
-                    webView.load(PrivilegedRequest(url: errorPageURL) as URLRequest)
-                } else if isNICErrorPageEnabled && (error.code == noInternetErrorCode) {
+                if isNativeErrorPageEnabled || (isNICErrorPageEnabled && (error.code == noInternetErrorCode)) {
```

Comment:
Both are dispatching the same action and loading the same `errorPageURL` so simplifying the `if` here too

Reply:
thanks for the clean up! It was probably a miss. What do you think about simplifying this code in general?

```
// Check if native error page feature flag
    // check the no Internet Connection Code (use guard)
    // dispatch actions

// else 
   /// ErrorPageHelper(certStore: profile.certStore).loadPage(error, forUrl: url, inWebView: webView)

Reply:
Ahh, that's what I was missing. I didn't knew if the `isNativeErrorPageEnabled` flag was like the "parent" flag that needed to be toggled before we can enable the sub-flag `isNICErrorPageEnabled`. Yes that makes sense to me!